rb/lib/selenium/webdriver/firefox/options.rb

```diff
@@ -140,6 +133,7 @@ def log_level=(level)
         def process_browser_options(browser_options)
           options = browser_options[KEY]
           options['binary'] ||= Firefox.path if Firefox.path
+          options['profile'] = options.delete(:profile) if profile
```

Comment:
Yeah, this is a weird one. And looking at it, I'm treating it differently from `Chrome::Options#extensions=`, which is wrong (I'll fix).

Essentially for these we don't want to use the default setter and getter that is autogenerated by `Options#self.set_capabilities`, since we want to validate correctness on construction and setter rather than when going to use it.

When each class was setting it as part of class construction, we could override with `undef`, but I think it is better to do it at construction, which lets me put the looping logic in the superclass. The tradeoff is that I have to pull these values out of `CAPABILITIES` and handle them special.

Since they aren't in `CAPABILITIES` they aren't processed in `#as_json` and they key stays as a `Symbol` instead of correctly getting translated into a `String` (the purpose of `#as_json`. Since `options[:profile]` is the same as `profile` it seemed like a shortcut to just use `profile`.

1. Either way I need to do the same behavior for `Chrome#extensions=`
2. Would it make more sense to just put the autogenerated accessor behavior back to current and use `undef` for these?
3. The other option is we could do the processing when `#generate_capabilities` gets called during `#create_bridge` instead of during constructor/setter

Reply:
Ok, I think I did the right thing. These will be tracked by instance variable rather than inside the options class since there's no reason to pass them around to superclass when they are only getting processed in their own subclass.